platforms/core-runtime/gradle-cli/src/main/java/org/gradle/launcher/cli/DefaultCommandLineActionFactory.java

```diff
@@ -166,25 +181,37 @@ private static class CommandLineParseFailureAction implements Action<ExecutionLi
         private final Exception exception;
         private final CommandLineParser parser;
 
-        public CommandLineParseFailureAction(CommandLineParser parser, Exception exception) {
+        private final List<String> args;
+
+        public CommandLineParseFailureAction(CommandLineParser parser, Exception exception, List<String> args) {
             this.parser = parser;
             this.exception = exception;
+            this.args = args;
         }
 
         @Override
         public void execute(ExecutionListener executionListener) {
             System.err.println();
             System.err.println(exception.getMessage());
-            showUsage(System.err, parser);
+            String suggestedTaskName = null;
```

Comment:
💅 Let's extract that into a separate function.

❓ Better yet, do we have a CLI arguments parser logic already somewhere? Can we reuse that?

Reply:
We only get here if parsing logic throws an exception